From 14d93b52ce538ba0446508473482fc4da91be3a2 Mon Sep 17 00:00:00 2001
From: Rahul Rameshbabu <rrameshbabu@nvidia.com>
Date: Tue, 11 Nov 2025 12:36:47 -0800
Subject: [PATCH] nvkms: Limit default maximum TMDS character rate to 340MHz
 instead of 165MHz

This will prevent pre-mature mode invalidation on a wider range of
monitors.

Signed-off-for-open-source-by: rrameshbabu

reviewed by lkrishnan, mtijanic
---
 src/nvidia-modeset/src/nvkms-dpy.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/src/nvidia-modeset/src/nvkms-dpy.c b/src/nvidia-modeset/src/nvkms-dpy.c
index bbdf4df50..afc9f3c03 100644
--- a/src/nvidia-modeset/src/nvkms-dpy.c
+++ b/src/nvidia-modeset/src/nvkms-dpy.c
@@ -914,8 +914,7 @@ void nvDpyProbeMaxPixelClock(NVDpyEvoPtr pDpyEvo)
 
                 if (pParsedEdid->valid) {
                     const NVT_EDID_INFO *pEdidInfo = &pParsedEdid->info;
-                    /* Default Maximum HDMI TMDS character rate is 165MHz. */
-                    NvU32 maxTmdsCharRate = 33;
+                    NvU32 maxTmdsCharRate = 68;
 
                     if (pEdidInfo->ext861.valid.H20_HF_VSDB &&
                         (pEdidInfo->hdmiForumInfo.max_TMDS_char_rate > 0)) {
